libpod/runtime_ctr.go

```diff
@@ -308,8 +309,15 @@ func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Contai
 
 		logrus.Debugf("Creating new volume %s for container", vol.Name)
 
+		// Get the uid/gid of the container user to create the volume
+		// with the correct uid/gid. See github.com/containers/libpod/issues/5698.
+		uid, gid, _, err := chrootuser.GetUser(ctr.state.Mountpoint, ctr.config.User)
```

Comment:
This is not safe - `ctr.state.Mountpoint` is unset at this point, because the container has not been mounted.

Reply:
If you pass a user by *name* instead of UID, this will break horribly and use the host's `/etc/passwd` instead of the container's.

Reply:
When I went over this with Sujil when he was working on it, we agreed that mounting the container this early in creation is prohibitively difficult, so he was investigating chown'ing the volume as part of copy-up.

Reply:
Eek. What shall we do?

Reply:
The best of a series of bad options was to do it as part of this block: https://github.com/containers/libpod/blob/f4c3b718eb22a161a897a6ed55d10f3a07e31aa8/libpod/container_internal.go#L1457-L1487

This only happens once, on first creation of the volume

Reply:
> If you pass a user by _name_ instead of UID, this will break horribly and use the host's `/etc/passwd` instead of the container's.

Can we support by _name_ here at all?

Reply:
If that's OK, it makes things *much* simpler

Reply:
can we move the chown later (container_internal_linux.go generateSpec)?  We already parse once the `/etc/passwd` and `/etc/group` files

Reply:
looking at it more in details, I think a possible solution is to have an operation like `ChownIfEmpty` on the volumes that we can use once we read `/etc/passwd` from the image